### ISSUE TITLE
HA for exitnode for auto selection on failure.

### ISSUE THREAD

**Issue by darshinimashar** [tailscale/tailscale#2598]
The customer is looking forward to the HA for the exit node, auto-selecting other exit nodes in case of failure.

**Comment by vysecurity** [comment]
Also will be good to show the ping to these servers in the CLI for status, and also on the right click / menu bar icons.

**Comment by nunnun** [comment]
+1
It would be better to have virtual exit-nodes for redundancy. For example,
Virtual Exit node (New York) -> exit-node-nyc1, exit-node-nyc2, exit-node-nyc3 (roundrobin)
Virtual Exit node (Tokyo) -> exit-node-tyo1, exit-node-tyo2, exit-node-tyo3 (roundrobin)

**Comment by mayakacz** [comment]
I'm exploring what we could do in this space, and have a few questions.
* Do you prefer to run exit nodes in an HA mode, but fixed in a region; or automatically fail over to the next closest exit node (which might be in a different region)?
* Should users be able to pick an exit node of their choice?
* If all exit nodes are deleted or not available, what should happen to traffic?

Thanks!

**Comment by nunnun** [comment]
@mayakacz
Thanks for the comments.

> Do you prefer to run exit nodes in an HA mode, but fixed in a region; or automatically fail over to the next closest exit node (which might be in a different region)?

We prefer fixed in a region (most reason why we use exit-node is that source IP address restriction for external services).
So it's better that the region (or HA-group) should be fixed.

> Should users be able to pick an exit node of their choice?

Should users be able to pick a specific exit node for HA group? -> No
Should users be able to pick an HA mode exit node? -> Yes

> If all exit nodes are deleted or not available, what should happen to traffic?

We think the traffic should be routed to default routes (non-tailscale network)

Thank you!

**Comment by vysecurity** [comment]
Agreed with above comment. Also, it’s be good to have automatic IP switching and load balancing between exit nodes as an optional feature to check in the UI. Might be useful for some enterprise applications too.

**Comment by tleerai** [comment]
> * Do you prefer to run exit nodes in an HA mode, but fixed in a region; or automatically fail over to the next closest exit node (which might be in a different region)?

Assuming these regions are user defined groups:
If a user explicitly selects a single group of exit nodes, I think it's unintuitive to start connecting them to an exit in another group. 

> * Should users be able to pick an exit node of their choice?

No, just one or possibly more group of exit nodes. By having the user select all the acceptable exit node groups they are willing to connect to, this should solve the intent issue, and render the first question irrelevant. 

> * If all exit nodes are deleted or not available, what should happen to traffic?

The ramifications of falling back to the default route may not be understood by the user nor safe for the user to decide, so this should be configured at the account or exit node group level. Possibly using an ACL to dictate who can reach the default route.

**Comment by PatrickHuetter** [comment]
+1

**Comment by Herklos** [comment]
+1

**Comment by dylan-upchurch** [comment]
Would love to see this feature also +1
